src/contrib/dependencyinjection/Akka.DependencyInjection/DependencyResolver.cs

```diff
@@ -111,15 +111,43 @@ public Props Props(Type type, params object[] args)
     /// </summary>
     public sealed class DependencyResolverExtension : ExtensionIdProvider<DependencyResolver>
     {
+        readonly IDependencyResolver _dependencyResolver;
+
+        public DependencyResolverExtension()
+        {
+        }
+
+        public DependencyResolverExtension(IDependencyResolver dependencyResolver)
```

Comment:
You don't want this code on the `ExtensionIdProvider` - you want it on the `DependencyResolver` extension itself. The  `ExtensionIdProvider` is just meant to be a unique identifier that the `ActorSystem` uses to uniquely identify each possible extension to ensure there's at most one instance of it per-`ActorSystem`.